nf_core/lint.py

```diff
@@ -103,21 +104,59 @@ def pf(file_path):
                 self.warned.append((1, "File not found: {}".format(files)))
 
 
+    def check_docker(self):
+        """minimal tests only"""
+        logging.debug('Checking Dockerfile')
+        fn = os.path.join(self.path, "Dockerfile")
+        with open(fn, 'r') as fh: content = fh.read()
+
+        # Implicitely also checks if empty.
+        if 'FROM ' in content:
+            self.passed.append((2, "Dockerfile check passed"))
+            return
+
+        self.failed.append((2, "Dockerfile check failed"))
+
+
     def check_licence(self):
         logging.debug('Checking licence file is MIT')
         for l in ['LICENSE', 'LICENSE.md', 'LICENCE', 'LICENCE.md']:
             fn = os.path.join(self.path, l)
             if os.path.isfile(fn):
-                if 'MIT' in open(fn).read():
-                    self.passed.append((2, "Licence check passed"))
+                with open(fn, 'r') as fh: content = fh.read()
+
+                # needs at least copyright, permission, notice and "as-is" lines
+                nl = content.count("\n")
+                if nl < 4:
+                    self.failed.append((2, "Number of lines too small for a valid MIT license file: {}".format(fn)))
                     return
-                else:
+
+                # determine whether this is indeed an MIT
+                # license. Most variations actually don't contain the
+                # string MIT Searching for 'without restriction'
```

Comment:
Hah, good catch! Sorry, I only checked one and the first line had MIT in.. This is better 👍